### DISCUSSION TITLE
Error loading workflow model file in VS Code extension

### DISCUSSION THREAD

**Discussion by gee4vee** [eclipse-glsp/glsp#681] [Q&A] [answered]
We are using the [example workflow node server](https://github.com/eclipse-glsp/glsp-server-node/tree/main/examples/workflow-server) and are working on making this example run as a VS Code extension just like the task list example. We have made a workflow client following the [task list project template](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-vscode/glsp-client) by copying the `tasklist-glsp` and `tasklist-vscode` sources and modifying them appropriately to load a workflow web view.

We started the server and the client successfully but when we try to open `example1.wf` we get this error in the server.

```
/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/.bin/ts-node ./glsp-server-node/src/index.ts
Debugger attached.
11:16:17 AM info -  [SocketServerLauncher] The GLSP server is ready to accept new client requests on port: 5007
[GLSP-Server]:Startup completed
/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/di/diagram-module.ts:314
        binding.add(ModelSourceChangedAction.KIND);
                                             ^
TypeError: Cannot read properties of undefined (reading 'KIND')
    at WorkflowDiagramModule.configureClientActions (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/di/diagram-module.ts:314:46)
    at /Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/di/diagram-module.ts:183:105
    at WorkflowDiagramModule.configureMultiBinding (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/di/glsp-module.ts:63:9)
    at WorkflowDiagramModule.configure (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/di/diagram-module.ts:183:14)
    at WorkflowDiagramModule.registry (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/di/glsp-module.ts:42:18)
    at Container.load (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/inversify/src/container/container.ts:110:27)
    at DefaultGlobalActionProvider.createDiagramContainer (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/actions/global-action-provider.ts:52:19)
    at /Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/actions/global-action-provider.ts:41:36
    at Map.forEach (<anonymous>)
    at new DefaultGlobalActionProvider (/Users/gvalenc/git/DevX/solution-builder-glsp-workflow/glsp-server-node/node_modules/@eclipse-glsp/server-node/src/actions/global-action-provider.ts:40:24)
Waiting for the debugger to disconnect...
```

Since there isn't a working workflow example with a node server, can we get some help to get this to work? We cannot use the Java server.

**Comment by gee4vee** [comment]
Looking into the stack trace, `ModelSourceChangedAction` does not exist in the latest version of `protocol`. It was renamed to `SourceModelChangedAction`.

<img width="1144" alt="image" src="https://user-images.githubusercontent.com/179807/172898790-f35ecfff-e030-462c-b96c-42bc455965b5.png">

**Reply by planger** [reply]
Yes, this has been recently renamed to `SourceModelChangedAction` to be in line with the terminology of a source model and the graphical model (see also https://github.com/eclipse-glsp/glsp/commit/a16967e126a6863784bd9d5342192585b53e1a40). Model source was an old term that we've used but it overlaps with a concept from sprotty. So we renamed it.

**Reply by gee4vee** [reply]
@planger Is this going to be fixed in `@eclipse-glsp/server-node`? That is where it is [trying to use](https://github.com/eclipse-glsp/glsp-server-node/blob/main/packages/server-node/src/di/diagram-module.ts#L21) `ModelSourceChangedAction`.

**Reply by planger** [reply]
Yes, there already is an [open PR](https://github.com/eclipse-glsp/glsp-server-node/pull/21) for it, but it hasn't been merged yet.
Since it now has been approved and went through the CI, I'll merge it now.

**Reply by gee4vee** [reply]
Thank you! When does the new NPM version usually arrive?

**Reply by planger** [reply]
Just landed: https://www.npmjs.com/package/@eclipse-glsp/server-node/v/0.9.0-next.b178eb2.24

**Comment by planger** [comment]
>Since there isn't a working workflow example with a node server, can we get some help to get this to work? We cannot use the Java server.

There is an example for the workflow diagram server based on the node server framework here: https://github.com/eclipse-glsp/glsp-server-node/tree/main/examples/workflow-server

**Reply by gee4vee** [reply]
Yes, I mentioned that one. We are using it. But there is no VS code integration client for workflow. That is what we are trying to build.

**Reply by planger** [reply]
Ah ok, I'm sorry, I was a bit confused :-)

**Comment by gee4vee** [comment, marked as answer]
The latest version of `server-node` fixes this. Thank you!

**Reply by planger** [reply]
That's great! :rocket:
 
Btw, if you are (planning to) use GLSP, we'd be very happy if you or your organization would be listed as a contributor/adopter at https://ecdtools.eclipse.org/adopters and you'd give us a star on Github. It is very easy [to get listed](https://ecdtools.eclipse.org/adopters/get-listed/), basically it is just opening an issue [here](https://gitlab.eclipse.org/eclipsefdn/it/api/eclipsefdn-project-adopters/-/issues/) with your logo. Thanks for considering!
